pbr/util.py

```diff
@@ -286,9 +286,57 @@ def point_on_field(cam_location, mask_path, env_info, num_points):
 
     # Check if environment map has field points, else set to origin
     if len(field_coords) > 0:
-        for ii in range(num_points):
+        while len(ground_points) < scene_config.num_robots:
             # Get random field point
             y, x = field_coords[rand.randint(0, field_coords.shape[0] - 1)]
+            yproj, xproj = project_to_ground(y, x, cam_location, img, env_info)
+
+            if any(
+                [
+                    (p[0] - xproj) ** 2 + (p[1] - yproj) ** 2
+                    < scene_config.radius_robot**2
+                    for p in ground_points
+                ]
+            ):
+                print("Skipping point")
+                continue
 
             ground_points.append(project_to_ground(y, x, cam_location, img, env_info))
+
     return ground_points
+
+
+# The function above will try and project the robots on to the ground
+# If the projection model is equidistant, the projection spawns the robots on top of one another on the origin
+# This function is a workaround to generate a set of random points on the field instead of just on the origin
+def generate_moves(field_meta, z_coord=0.3):
+
+    field_dims = (
+        field_meta["length"] + 2 * field_meta["border_width"],
+        field_meta["width"] + 2 * field_meta["border_width"],
+    )
+
+    # Use the field dimensions to generate a set of moves, mainly for the robots
+    abs_x, abs_y, _ = field_dims
+
+    world_points = []
+
+    while len(world_points) < scene_config.num_robots:
+        # Get random field point
+        point = np.random.uniform(
+            low=(-abs_x / 2, -abs_y / 2), high=(abs_x / 2, abs_y / 2)
+        )
+
+        if any(
+            [
+                (p[0] - point[0]) ** 2 + (p[1] - point[1]) ** 2
+                < scene_config.robot_radius**2
+                for p in world_points
+            ]
+        ):
+            print("Skipping point")
```

Comment:
Does this spam?

Reply:
yes

Reply:
I use black as my default formatter. Should we make another pr that adds a .vscode folder?

Reply:
Also, I just think its a nice debugging tool so that if somebody changes the value and they somehow end up in an infinite loop they know that their script is stuck there

Reply:
yeah, but it would be good to have a way to toggle it on/off if you want to see other output. Maybe a debug config value. 
the formatting can be addressed in another PR. that can add vs code settings and a check for formatting.